performance/src/performance/timer.cpp

```diff
@@ -11,6 +11,9 @@
 // System include(s).
 #include <algorithm>
 
+// Nividia tool extensions library
+#include "nvtx3/nvToolsExt.h"
```

Comment:
You will need to "protect" this code from being seen in cases where CUDA is not available. That is the "meat" of this PR in the end.

In the CMake configuration you should have something like:

```cmake
find_package(CUDAToolkit)
if (CUDAToolkit_FOUND)
   target_link_libraries( traccc_performance PRIVATE CUDA::cudart )
   target_compile_definitions( traccc_performance PRIVATE TRACCC_HAVE_NVTX )
endif()
```

Then you can surround all these statements in `timer.cpp` like:

```c++
#ifdef TRACCC_HAVE_NVTX
#  include <nvtx3/nvToolsExt.h>
#endif  // TRACCC_HAVE_NVTX
```

Reply:
Made this adjustment